certified-connectors/Asite (KSA)/apiDefinition.swagger.json

```diff
@@ -0,0 +1,1163 @@
+        "operationId": "FILE_DOWNLOAD_BY_URL",
+        "parameters": [
+          {
+            "name": "downloadUrl",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "Select 'DirectLink' metadata to download a file",
+            "x-ms-summary": "URL Selection"
+          }
+        ]
+      }
+    },
+    "/triggerList": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "OK, Success",
+            "schema": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "triggerId": {
+                    "type": "integer",
+                    "format": "int32",
+                    "description": "Trigger ID"
+                  },
+                  "triggerName": {
+                    "type": "string",
+                    "description": "Trigger Name"
+                  }
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "BAD REQUEST, Invalid request"
+          }
+        },
+        "summary": "List of configured Triggers from Asite Platform",
+        "description": "List of configured Triggers from Asite Platform",
+        "operationId": "ASITE_TRIGGER_LIST",
+        "x-ms-visibility": "internal",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "projectId",
+            "in": "query",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "ruleContext",
+            "in": "query",
+            "required": true,
+            "type": "integer"
+          },
+          {
+            "name": "namespaceId",
+            "in": "query",
+            "required": false,
+            "type": "integer"
+          }
+        ]
+      }
+    },
+    "/deleteAsitePullDataWebhook/{id}/{nameOfClass}": {
+      "delete": {
+        "responses": {
+          "200": {
+            "description": "OK, Success"
+          }
+        },
+        "summary": "Delete configured trigger",
+        "description": "Delete configured trigger",
+        "operationId": "DELETE_TRIGGER_WEBHOOK",
+        "x-ms-visibility": "internal",
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "x-ms-url-encoding": "single",
+            "type": "string"
+          },
+          {
+            "name": "nameOfClass",
+            "in": "path",
+            "required": true,
+            "x-ms-url-encoding": "single",
+            "type": "string"
+          }
+        ]
+      }
+    },
+    "/getFolderAttributes": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "OK, Success",
+            "schema": {}
```

Comment:
@asitedeveloper Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger. **This applies to all the empty schemas in the swagger.**

Reply:
@sriyen-msft We have sent the dynamic schema in response to our API call. Hence we have set schema as empty in swagger JSON. The dynamic schema that we are sending in API response is as below. The fields in this schema is fully dynamic and based on values selected for an API call. So it can be different for different values of the respective parameters. So as it is not fixed, we have kept the schema as empty.

```
{
	"type": "object",
	"properties": {
		"DocStatus": {
			"type": "string",
			"title": "DocStatus"
		},
		"PublisherOrg": {
			"type": "string",
			"title": "PublisherOrg"
		}
	}
}
```